<a href="https://colab.research.google.com/github/eric-pding/RSM8413_Group_Assignments/blob/main/RSM8413_Group_Assignment_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# RSM8413: Group Assignment 4

## Part I: Data Preprocessing

In [1]:
from sklearn.cluster import KMeans
import plotly.express as px
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
# Reading in the Dataset
train = 'https://raw.githubusercontent.com/eric-pding/RSM8413_Group_Assignments/main/CensusCanada2021Training.csv'
test = 'https://raw.githubusercontent.com/eric-pding/RSM8413_Group_Assignments/main/CensusCanada2021Test.csv'

data_train = pd.read_csv(train)
data_test = pd.read_csv(test)

In [7]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5371 entries, 0 to 5370
Data columns (total 18 columns):
 #   Column                                                                     Non-Null Count  Dtype  
---  ------                                                                     --------------  -----  
 0   Total Population                                                           5371 non-null   int64  
 1   Total Households                                                           5297 non-null   float64
 2   Median Household Income (Current Year $)                                   5245 non-null   float64
 3   Total Households For Period Of Construction                                5297 non-null   float64
 4   Total Households For Period Of Construction Built Before 1961              5297 non-null   float64
 5   Total Households For Period Of Construction Built Between 1961 And 1980    5297 non-null   float64
 6   Total Households For Period Of Construction Built Betwee

In [8]:
data_train.head()

Total Population  Total Households  \
0              3088            1235.0   
1              5889            1665.0   
2              8090            2775.0   
3              4074            1775.0   
4              3642            1385.0   

   Median Household Income (Current Year $)  \
0                                  107000.0   
1                                  111000.0   
2                                  105000.0   
3                                   65000.0   
4                                  101000.0   

   Total Households For Period Of Construction  \
0                                       1235.0   
1                                       1665.0   
2                                       2775.0   
3                                       1775.0   
4                                       1385.0   

   Total Households For Period Of Construction Built Before 1961  \
0                                               25.0               
1                                                0.0               
2                                              105.0               
3                                              210.0               
4                                               10.0               

   Total Households For Period Of Construction Built Between 1961 And 1980  \
0                                             1130.0                         
1                                                0.0                         
2                                              150.0                         
3                                             1100.0                         
4                                              190.0                         

   Total Households For Period Of Construction Built Between 1981 And 190  \
0                                               50.0                        
1                                                0.0                        
2                                              240.0                        
3                                              260.0                        
4                                             1010.0                        

   Total Households For Period Of Construction Built Between 1991 And 2000  \
0                                               15.0                         
1                                               15.0                         
2                                             1210.0                         
3                                               85.0                         
4                                              165.0                         

   Total Households For Period Of Construction Built Between 2001 And 2005  \
0                                               10.0                         
1                                               20.0                         
2                                              555.0                         
3                                               35.0                         
4                                                0.0                         

   Total Households For Period Of Construction Built Between 2006 And 2010  \
0                                               10.0                         
1                                              375.0                         
2                                              165.0                         
3                                               45.0                         
4                                                0.0                         

   Total Households For Period Of Construction Built Between 2011 And 2015  \
0                                                0.0                         
1                                              315.0                         
2                                              240.0                         
3                                               30.0                         
4                                 

**Insights from the Dataset**


*   Missing Values in many of the columns (5371 in total but most had 5297)
*   Naming of the column (Total Households For Period Of Construction Built Between 1981 And 190)?
*   There is a column named "Tenure Renter" in the excel file that has no entries; however, when loading the data, it is not in the dataset




In [11]:
# Renaming the Misnamed Column
data_train.rename(columns= {"Total Households For Period Of Construction Built Between 1981 And 190":
                            "Total Households For Period Of Construction Built Between 1981 And 1990"}, inplace = True)
data_test.rename(columns= {"Total Households For Period Of Construction Built Between 1981 And 190":
                            "Total Households For Period Of Construction Built Between 1981 And 1990"}, inplace = True)

data_train.head()

Total Population  Total Households  \
0              3088            1235.0   
1              5889            1665.0   
2              8090            2775.0   
3              4074            1775.0   
4              3642            1385.0   

   Median Household Income (Current Year $)  \
0                                  107000.0   
1                                  111000.0   
2                                  105000.0   
3                                   65000.0   
4                                  101000.0   

   Total Households For Period Of Construction  \
0                                       1235.0   
1                                       1665.0   
2                                       2775.0   
3                                       1775.0   
4                                       1385.0   

   Total Households For Period Of Construction Built Before 1961  \
0                                               25.0               
1                                                0.0               
2                                              105.0               
3                                              210.0               
4                                               10.0               

   Total Households For Period Of Construction Built Between 1961 And 1980  \
0                                             1130.0                         
1                                                0.0                         
2                                              150.0                         
3                                             1100.0                         
4                                              190.0                         

   Total Households For Period Of Construction Built Between 1981 And 1990  \
0                                               50.0                         
1                                                0.0                         
2                                              240.0                         
3                                              260.0                         
4                                             1010.0                         

   Total Households For Period Of Construction Built Between 1991 And 2000  \
0                                               15.0                         
1                                               15.0                         
2                                             1210.0                         
3                                               85.0                         
4                                              165.0                         

   Total Households For Period Of Construction Built Between 2001 And 2005  \
0                                               10.0                         
1                                               20.0                         
2                                              555.0                         
3                                               35.0                         
4                                                0.0                         

   Total Households For Period Of Construction Built Between 2006 And 2010  \
0                                               10.0                         
1                                              375.0                         
2                                              165.0                         
3                                               45.0                         
4                                                0.0                         

   Total Households For Period Of Construction Built Between 2011 And 2015  \
0                                                0.0                         
1                                              315.0                         
2                                              240.0                         
3                                               30.0                         
4                           